RUFAS/input_manager.py

```diff
@@ -53,144 +77,368 @@ def _load_metadata(self, metadata_path: str = "input/example_metadata.json") ->
+            false_counter = 0
+            for nested_key in variable_properties.keys():
+                element_hierarchy.append(nested_key)
+                child_status = self._validate_element(element_hierarchy, properties_blob_key,
+                                                      input_data, eager_termination)
+                if eager_termination and not child_status:
+                    return False
+                element_path = ".".join(element_hierarchy)
+                children_status[element_path] = child_status
+                if not child_status:
+                    om.add_warning("Invalid nested element found", f"{element_path}", info_map)
+                    false_counter += 1
+                element_hierarchy.remove(nested_key)
+            is_valid = false_counter == 0
+            if is_valid:
+                return True
+            else:
+                invalid_child_vars = [key for key, value in children_status.items() if value is False]
+                om.add_warning("Invalid nested element(s) found",
+                               f"{invalid_child_vars=}",
+                               info_map)
+                return False
+        else:
+            var_name = element_hierarchy[-1]
+            try:
+                input_data_value = reduce(lambda d, key: d[key], element_hierarchy, input_data)
+            except KeyError:
+                raise KeyError(f"Key {var_name} not found in input data")
 
-        # TODO in validate_element fun branch
-        # where element is found to be invalid, place this warning:
-        # om.add_warning("Invalid data", f"Invalid data found: {key=}; {value=}", info_map)
-        pass
+            type_validation_dict = {"string": self._validate_string_type_element,
+                                    "number": self._validate_num_type_element,
+                                    "array": self._validate_array_type_element,
+                                    "bool": self._validate_bool_type_element, }
+
+            try:
+                validity_checker = type_validation_dict[var_type]
+                is_valid = validity_checker(variable_properties, var_name, input_data_value)
+            except KeyError:
+                raise KeyError(f"Invalid type {var_type}: Element must be type number, array, string, or bool")
+
+            counter_dict["total_elements"] += 1
+            if is_valid:
+                counter_dict["valid_elements"] += 1
+                return True
+            else:
+                is_fixed = self._fix_data(variable_properties, element_hierarchy, input_data)
+                if is_fixed:
+                    counter_dict["fixed_elements"] += 1
+                else:
+                    counter_dict["invalid_elements"] += 1
+                return is_fixed
 
-    def _fix_data(self, key: str, value: Any) -> bool:
+    def _validate_array_type_element(self, variable_properties: Dict[str, Any], var_name: str,
```

Comment:
what about?
```suggestion
    def _array_type_validator(self, variable_properties: Dict[str, Any], var_name: str,
```
and for other similar functions.

Reply:
I like this!